internal/services/oracledatabase/exadata_infrastructure_resource.go

```diff
@@ -0,0 +1,387 @@
+				}
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ExadataInfraResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+
+			client := metadata.Client.OracleDatabase.OracleDatabaseClient.CloudExadataInfrastructures
+			id, err := cloudexadatainfrastructures.ParseCloudExadataInfrastructureID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ExadataInfraResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding err: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving exists when updating: +%v", *id)
+			}
+			if existing.Model == nil && existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving as nil when updating for %v", *id)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				update := &cloudexadatainfrastructures.CloudExadataInfrastructureUpdate{
+					Tags: tags.Expand(model.Tags),
+				}
+				err = client.UpdateThenPoll(ctx, *id, *update)
+				if err != nil {
+					return fmt.Errorf("updating %s: %v", id, err)
+				}
+			} else if metadata.ResourceData.HasChangesExcept("tags") {
+				return fmt.Errorf("only `tags` currently support updates")
+			}
+			return nil
+		},
+	}
+}
+
+func (ExadataInfraResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			id, err := cloudexadatainfrastructures.ParseCloudExadataInfrastructureID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := metadata.Client.OracleDatabase.OracleDatabaseClient.CloudExadataInfrastructures
+			result, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(result.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return err
+			}
+
+			if result.Model == nil {
+				return fmt.Errorf("retrieving %s got nil model", id)
+			}
+			var output ExadataInfraResourceModel
+
+			output.CustomerContacts = FlattenCustomerContacts(result.Model.Properties.CustomerContacts)
+			output.Name = pointer.ToString(result.Model.Name)
+			output.Location = result.Model.Location
+			output.Zones = result.Model.Zones
+			output.ResourceGroupName = id.ResourceGroupName
+			output.Tags = utils.FlattenPtrMapStringString(result.Model.Tags)
+			prop := result.Model.Properties
+			output.ComputeCount = pointer.From(prop.ComputeCount)
+			output.DisplayName = prop.DisplayName
+			output.StorageCount = pointer.From(prop.StorageCount)
+			output.Shape = prop.Shape
+			output.MaintenanceWindow = FlattenMaintenanceWindow(prop.MaintenanceWindow)
```

Comment:
minor issue but to match other resources we should inline this/do an assingment

```
output := ExadataInfraResourceModel {
CustomerContacts: FlattenCustomerContacts(result.Model.Properties.CustomerContacts)
Name: pointer.ToString(result.Model.Name)
...
```
and such vs going outout.x

Reply:
Understood, I've updated the code.